src/Rule.php

```diff
@@ -2135,6 +2137,8 @@ public function showRulePreviewResultsForm($input, $params)
         $entries = [
             [
                 'action' => __('Validation'),
+                'itemtype' => static::class,
+                'id' => $this->fields['id'],
```

Comment:
The itemtype and id here don't make sense. Maybe the `data-itemtype` and `data-id` attributes in the datatable rows should not be added at all when `itemype` and `id` are missing from the entry.

Reply:
I've reverted this change, and added a condition template side; this seems better.